tests/test_build_latex.py

```diff
@@ -942,3 +960,33 @@ def test_latex_table(app, status, warning):
     # longtable having both :widths: and problematic cell
     table = tables['longtable having both :widths: and problematic cell']
     assert ('\\begin{longtable}{|\\X{30}{100}|\\X{70}{100}|}' in table)
+
+
+@pytest.mark.skipif(docutils.__version_info__ < (0, 13),
+                    reason='docutils-0.13 or above is required')
+@pytest.mark.sphinx('latex', testroot='latex-table',
+                    confoverrides={'latex_elements': {'table_align': 'H'}})
+def test_latex_table_align(app, status, warning):
+    app.builder.build_all()
+    result = (app.outdir / 'test.tex').text(encoding='utf8')
+    tables = {}
+    for chap in re.split(r'\\section{', result)[1:]:
+        sectname, content = chap.split('}', 1)
+        tables[sectname] = content.strip()
+
+    # table having :align: option (tabulary)
+    table = tables['table having :align: option (tabulary)']
+    assert ('\\begin{table}[H]\\centering\n'
+            '\\noindent\\begin{tabulary}{\\linewidth}{|L|L|}\n' in table)
+    assert ('\\hline\n\\end{tabulary}\\end{table}' in table)
+
+    # table having :align: option (tabular)
+    table = tables['table having :align: option (tabular)']
+    assert ('\\begin{table}[H]\\begin{flushleft}\n'
+            '\\noindent\\begin{tabular}{|\X{30}{100}|\X{70}{100}|}\n' in table)
+    assert ('\\hline\n\\end{tabular}\\end{flushleft}\\end{table}' in table)
+
+    # longtable having :align: option (no effect)
```

Comment:
why no effect ? I have not tested but it should work, except if table is already `\linewidth` wide.